include/nbl/builtin/hlsl/cpp_compat/intrinsics.hlsl

```diff
@@ -20,10 +20,19 @@ namespace hlsl
 {
 
 template<typename Integer>
-int bitCount(NBL_CONST_REF_ARG(Integer) val)
+inline int bitCount(NBL_CONST_REF_ARG(Integer) val)
 {
 #ifdef __HLSL_VERSION
+	if (sizeof(Integer) == 8u)
+	{
+		uint32_t lowBits = val;
+		uint32_t highBits = val >> 32u;
```

Comment:
I think you need to bitcast `val` to unsigned int first, right shifting a signed int has weird behaviours w.r.t the sign bit